src/dns/aardvark.rs

```diff
@@ -19,15 +20,38 @@ const SYSTEMD_CHECK_PATH: &str = "/run/systemd/system";
 const SYSTEMD_RUN: &str = "systemd-run";
 const AARDVARK_COMMIT_LOCK: &str = "aardvark.lock";
 
-#[derive(Clone, Debug)]
+/// For better safety we wrap &str in our own custom type here
+/// where we can enforce that the caller can only construct it
+/// via the TryFrom trait. With that we can guarantee via the
+/// type system that we never get invalid chars here when we
+/// write the config file.
+#[derive(Debug)]
+pub struct SafeString<'a>(&'a str);
+
+impl<'a> TryFrom<&'a str> for SafeString<'a> {
+    type Error = &'static str;
+
+    fn try_from(value: &'a str) -> std::result::Result<Self, Self::Error> {
+        if value.is_empty() {
+            return Err("name is empty");
+        }
+        if value.contains([',', '\n', ' ']) {
```

Comment:
Feels like this is nowhere near aggressive enough. A valid DNS hostname contains letters, numbers, and hyphens only. An alias containing, for example, `#` should still break us. I know we dislike regexes but with something as restrictive as this I think it makes sense to use one.

Reply:
Well yes, but so far my only focus was to ensure we do not generate an invalid file. Everything else should get handled by aardvark-dns IMO. I didn't find a clear way where exactly is described  what works and what not. I know the actual spec is rather strict but I worry that docker might allow more?